src/main/java/com/ibm/stocator/fs/common/ObjectStoreGlobber.java

```diff
@@ -165,7 +165,7 @@ private String authorityFromPath(Path path) throws IOException {
           candidate.setPath(new Path(candidate.getPath(), component));
         }
       } else {
-        globFilter = new ObjectStoreGlobFilter("*");
+        globFilter = new ObjectStoreGlobFilter(components.get(0));
```

Comment:
@talARL the else statement executes to "true" when components == null or has 0 size. Your addition will throw null pointer exception.